packages/desktop/src/index.js

```diff
@@ -399,6 +401,34 @@ class Interpreter {
           break;
         case "dangerous_inner_html":
           node.innerHTML = value;
+          break;
+        case "href":
+
+          if (node.tagName == "A") {
+            // open the <a> tag in browser
+            node.setAttribute("browser-href", value);
+            node.setAttribute("href", "#");
+
+            node.addEventListener("click", function(event) {
+
+              const target = event.target;
+              const real_id = target.getAttribute(`dioxus-id`);
+
+              if (target.getAttribute("browser-open") != "false") {
+                rpc.call("browser_open", {
+                  mounted_dom_id: parseInt(real_id),
+                  href: target.getAttribute("browser-href"),
+                });
+              } else {
+                window.location = target.getAttribute("browser-href");
+              }
+
+            })
+
+          } else {
+            node.setAttribute(name, value);
+          }
+
```

Comment:
This is a good start, but I'd like to see this code moved into the event handler instead of setattribute.

We use listener multiplexing - so there's only one `click` handler for the whole app. Adding more handlers to the app manually might interfere with our global handling.

Instead of having this code *here*, we need it to either go in the Rust side where we receive events from webview, or in the `AddEventListener` code where we create the global listeners.

Additionally, instead of the "browser-open" attribute, we should just look at the URL itself and make the decision automatically. If it starts with a protocol (ie wry:// or dioxus:// - or anything that's not http:// or https://) then we want the url to move through the handler infrastructure.

Also - any `onclick` in Desktop should have "default prevented" - that way we don't need to mess with `window.location`. By calling "prevent default", the history will be preserved properly, so we can still provide links into a desktop app from the outside.

To recap:
- Move rpc.call into the "add event listener"
- Always call "prevent-default" for `onclick` events in desktop
- Parse URL instead of "browser-open" flag